lib/matplotlib/tests/test_datetime.py

```diff
@@ -20,11 +20,61 @@ def test_annotate(self):
         fig, ax = plt.subplots()
         ax.annotate(...)
 
-    @pytest.mark.xfail(reason="Test for arrow not written yet")
     @mpl.style.context("default")
     def test_arrow(self):
-        fig, ax = plt.subplots()
-        ax.arrow(...)
+        mpl.rcParams["date.converter"] = 'concise'
+        fig, (ax1, ax2, ax3) = plt.subplots(3, 1,
+                                            constrained_layout=True,
+                                            figsize=(10, 12))
+
+        start_date = datetime.datetime(2023, 1, 1)
+        dates = [start_date + datetime.timedelta(days=i) for i in range(31)]
+        y_values = range(31)
+
+        ax1.plot(dates, y_values, marker='o')
+        ax1.arrow(x=dates[10],
+                  y=10,
+                  dx=10,
+                  dy=10,
+                  head_width=1,
+                  head_length=1,
+                  linewidth=3,
+                  facecolor='red',
+                  edgecolor='red',
+                  transform=ax1.transData)
```

Comment:
specifying the transform as `transData` is not necessary, as that is the default.